application/core/LimeMailer.php

```diff
@@ -585,11 +585,10 @@ public function sendMessage()
             if (strpos($this->Body, "<html>") === false) {
                 $this->Body = "<html>" . $this->Body . "</html>";
             }
-            $this->msgHTML($this->Body, App()->getConfig("publicdir")); // This allow embedded image if we remove the servername from image
-            if (empty($this->AltBody)) {
-                $html = new \Html2Text\Html2Text($this->Body);
-                $this->AltBody = $html->getText();
-            }
+            $this->msgHTML($this->Body, App()->getConfig("publicdir"), function($html) {
+                return (new \Html2Text\Html2Text($html))->getText();
+            }); // This allow embedded image if we remove the servername from image
+            // TODO: Original AltBody is overwritten by msgHTML. Do we need to set it again if there was one?
```

Comment:
Before we leave https://github.com/LimeSurvey/LimeSurvey/blob/898dec3ad68a0186f815eed8c5f6dd2ca7f4e4eb/vendor/phpmailer/phpmailer/src/PHPMailer.php#L4387

And if it's empty we use `\Html2Text\Html2Text` 

Now we force `\Html2Text\Html2Text` ?

Maybe extend html2text function on LimeMailer directly is better ? and more understandable ?

Reply:
> Maybe extend html2text function on LimeMailer directly is better ? and more understandable ?

Not sure why it would be more understandable.
As it is now, we are using the intended way of overriding the standard converter and use a custom one.

> Now we force \Html2Text\Html2Text ?

Same way as we were trying to force it before.

Reply:
> Not sure why it would be more understandable. As it is now, we are using the intended way of overriding the standard converter and use a custom one.

Because html2text phpMailer function wads never used. Since we already extend it : more clear to replace the function. Really more clear